Mathlib/Data/Set/Function.lean

```diff
@@ -562,14 +562,21 @@ theorem MapsTo.mem_iff (h : MapsTo f s t) (hc : MapsTo f sᶜ tᶜ) {x} : f x 
 
 section
 
-variable (t f)
+variable (s t f)
 
 theorem range_restrictPreimage : range (t.restrictPreimage f) = Subtype.val ⁻¹' range f := by
   delta Set.restrictPreimage
   rw [MapsTo.range_restrict, Set.image_preimage_eq_inter_range, Set.preimage_inter,
     Subtype.coe_preimage_self, Set.univ_inter]
 #align set.range_restrict_preimage Set.range_restrictPreimage
 
+theorem image_restrictPreimage :
```

Comment:
Somebody said that code review is about asking questions to the code. I have a question:
can the proof of `range_restrictPreimage` be simplified using this lemma?

(Perhaps the answer is "no, it cannot" or "proving it this way is not simpler" - in these case, one shouldn't do it.)

Reply:
I simplified the proof of `range_restrictPreimage` using `image_restrictPreimage`. The new proof is indeed way simpler!
However, the proof I wrote seems not to be the more effective one, but I didn't find a way to get rid of the extra `rfl` at the end.

Reply:
Fixed in suggestion below.